src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -463,16 +462,11 @@ public int getPotentialMaxCitizens()
      */
     private double maxCitizensFromResearch()
     {
-        double max = 25;
-        final AdditionModifierResearchEffect effect = colony.getResearchManager().getResearchEffects().getEffect(CAP, AdditionModifierResearchEffect.class);
-        if (effect != null)
+        final double max = 25 + colony.getResearchManager().getResearchEffects().getEffectValue(CITIZEN_CAP);
+        // TODO research data rework
+        if (max >= MineColonies.getConfig().getServer().maxCitizenPerColony.get())
```

Comment:
What is this todo for?

Reply:
No clue.  Was present before I started, and I don’t think I finished it.  Presumably want to handle broader max citizen caps, but I don’t have enough experience with big colonies to be comfortable changing that behavior

Reply:
You can remove it then.